modules/monitoring/controllers/pnp.php

```diff
@@ -49,9 +49,32 @@ public function pnp_default()
 		if ($pnp_path != '') {
 			$source = intval($this->input->post('source', false));
 			$view = intval($this->input->post('view', false));
+			$settings = json_encode(array("source"=>$source, "view"=>$view));
 
-			Ninja_setting_Model::save_page_setting('source', $pnp_path.'/image?'.$param, $source);
-			Ninja_setting_Model::save_page_setting('view', $pnp_path.'/image?'.$param, $view);
+			Ninja_setting_Model::save_page_setting('default_graph', $pnp_path.'/image?'.$param, $settings);
 		}
 	}
+
+	/**
+	 * Get graph setting for a specific host/service
+	 */
+	public function get_pnp_default()
```

Comment:
Maybe use a more specific function name? This is easily confused with the already existing `pnp_default` function above. E.g. just `get_pnp_default_settings` would be better.

Reply:
Sure, why not.